azure/services/subnets/spec_test.go

```diff
@@ -20,291 +20,143 @@ import (
-			},
-			expectedError: "",
-		},
-		{
-			name:     "get parameters for subnet with multiple cidr blocks",
-			spec:     &fakeSubnetMultipleCidrSpec,
-			existing: nil,
-			expect: func(g *WithT, result interface{}) {
-				g.Expect(result).To(Equal(fakeSubnetMultipleCidrParams))
-			},
-			expectedError: "",
-		},
-		{
-			name:     "error vnet is not managed but subnet is missing",
-			spec:     &fakeSubnetSpecNotManaged,
-			existing: nil,
-			expect: func(g *WithT, result interface{}) {
-				g.Expect(result).To(BeNil())
-			},
-			expectedError: "custom vnet was provided but subnet my-subnet-1 is missing",
-		},
-		{
-			name:     "vnet is not managed and subnet is present",
-			spec:     &fakeSubnetSpecNotManaged,
-			existing: fakeSubnetNotManaged,
-			expect: func(g *WithT, result interface{}) {
-				g.Expect(result).To(BeNil())
-			},
-			expectedError: "",
-		},
-		{
-			name:     "error vnet is not managed but ipv6 subnet is missing",
-			spec:     &fakeIpv6SubnetSpecNotManaged,
-			existing: nil,
-			expect: func(g *WithT, result interface{}) {
-				g.Expect(result).To(BeNil())
-			},
-			expectedError: "custom vnet was provided but subnet my-ipv6-subnet is missing",
-		},
-		{
-			name:     "vnet is not managed and ipv6 subnet is present",
-			spec:     &fakeIpv6SubnetSpecNotManaged,
-			existing: fakeIpv6SubnetNotManaged,
-			expect: func(g *WithT, result interface{}) {
-				g.Expect(result).To(BeNil())
-			},
-			expectedError: "",
-		},
-	}
-	for _, tc := range testcases {
-		tc := tc
-		t.Run(tc.name, func(t *testing.T) {
-			g := NewWithT(t)
-			t.Parallel()
-
-			result, err := tc.spec.Parameters(context.TODO(), tc.existing)
-			if tc.expectedError != "" {
-				g.Expect(err).To(HaveOccurred())
-				g.Expect(err).To(MatchError(tc.expectedError))
-			} else {
-				g.Expect(err).NotTo(HaveOccurred())
-			}
-			tc.expect(g, result)
-		})
-	}
-}
-
-func TestSubnetSpec_shouldUpdate(t *testing.T) {
-	type fields struct {
-		Name              string
-		ResourceGroup     string
-		SubscriptionID    string
-		CIDRs             []string
-		VNetName          string
-		VNetResourceGroup string
-		IsVNetManaged     bool
-		RouteTableName    string
-		SecurityGroupName string
-		Role              infrav1.SubnetRole
-		NatGatewayName    string
-		ServiceEndpoints  infrav1.ServiceEndpoints
-	}
-	type args struct {
-		existingSubnet armnetwork.Subnet
-	}
 	tests := []struct {
-		name   string
-		fields fields
-		args   args
-		want   bool
+		name     string
+		spec     *SubnetSpec
+		existing *asonetworkv1.VirtualNetworksSubnet
+		expected *asonetworkv1.VirtualNetworksSubnet
 	}{
 		{
-			name: "subnet should not be updated when the VNet is not managed",
-			fields: fields{
-				Name:           "my-subnet",
-				ResourceGroup:  "my-rg",
-				SubscriptionID: "123",
-				IsVNetManaged:  false,
-			},
-			args: args{
-				existingSubnet: armnetwork.Subnet{
-					Name: ptr.To("my-subnet"),
+			name: "no existing subnet",
```

Comment:
It seems like we're just testing the case where there is an existing subnet and one where there isn't. Do you think it's worth preserving any of the other test cases from before, or would they not really be checking any new control flow?

Reply:
`Parameters` ultimately has fewer responsibilities now than it had before (doesn't error for unmanaged vnet, doesn't diff with existing subnet) and it seemed like some of the old cases were redundant, so I think these two cases provide adequate coverage.